### PR TITLE
Add Telemetry to Taiga

### PR DESCRIPTION
**Pull request by yamila-moreno**

### CODE REVIEW

File: taiga/telemetry/services.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+    pd = {}
+
+    # number of projects
+    pd['tt_projects'] = Project.objects.count()
+
+    # number of private projects
+    pd['tt_projects_private'] = Project.objects.filter(
+            is_private=True
+        ).count(),
```
Comment (bameda) [inline taiga/telemetry/services.py:82]: The `,` at the end of this line generate a tuple (a list in the json request sent to the telemetry service)

File: taiga/telemetry/services.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+            total_custom_fields=0
+        ).count()
+
+    # number of users
+    pd['tt_users'] = User.objects.count()
+
+    # number of active users
+    pd['tt_users_active'] = User.objects.filter(
+            is_active=True
+        ).count()
```
Comment (bameda) [inline taiga/telemetry/services.py:150]: Same here

File: taiga/telemetry/services.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+            total_custom_fields=Count('epiccustomattributes', distinct=True) + \
+                                Count('issuecustomattributes', distinct=True) + \
+                                Count('taskcustomattributes', distinct=True) + \
+                                Count('userstorycustomattributes', distinct=True)
+        ).exclude(
+            total_custom_fields=0
+        ).count()
+
+    # number of users
+    pd['tt_users'] = User.objects.count()
```
Comment (bameda) [inline taiga/telemetry/services.py:145]: add `excludes(is_system=True)` to exclude system users (like github, gitlab... integration users)

**Review by bameda — CHANGES_REQUESTED** [review]

File: taiga/telemetry/services.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+#     RawSQL also aggregates the SQL to the `group by` clause which defeats the purpose of adding it to an Annotation.
+#     """
+#     def get_group_by_cols(self):
+#         return []
+# # The Query
+# q = MyModel.objects.values('some_fk_id').annotate(
+#     avg_duration=Avg('duration'),
+#     perc_90_duration=RawAnnotation('percentile_disc(%s) WITHIN GROUP (ORDER BY duration)', (0.9,)),
+# )
+
```
[the remark this answers is not in the document]
Comment (yamila-moreno) [reply taiga/telemetry/services.py:58]: fixed!

File: taiga/telemetry/services.py (hunk 1 above)
Comment (yamila-moreno) [reply taiga/telemetry/services.py:82]: fixed!

File: taiga/telemetry/services.py (hunk 2 above)
Comment (yamila-moreno) [reply taiga/telemetry/services.py:150]: you mean the comma? or something different?

File: taiga/telemetry/services.py (hunk 3 above)
Comment (yamila-moreno) [reply taiga/telemetry/services.py:145]: fixed!

File: taiga/telemetry/services.py (hunk 2 above)
Comment (bameda) [reply taiga/telemetry/services.py:150]: No, I'd use ` excludes(is_system=True)` here too.

Comment (yamila-moreno) [reply taiga/telemetry/services.py:143]: fixed!

**Review by bameda — APPROVED** [review]

**Pull request merged**